### PR TITLE
[CRUD Suite] Delete extra subnet before deleting cluster

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: integration/tests/crud/creategetdelete_test.go [hunk 1]
```diff
@@ -1083,6 +1083,13 @@ var _ = Describe("(Integration) Create, Get, Scale & Delete", func() {
 })
 
 var _ = SynchronizedAfterSuite(func() {}, func() {
+	// before deleting the cluster, first delete the additioanl subnet
```
Comment (cPu1) [inline integration/tests/crud/creategetdelete_test.go:1086]:
```suggestion
	// before deleting the cluster, first delete the additional subnet
```

**Review by cPu1 — APPROVED** [review]

**Comment by cPu1** [comment]
There are some additional commits in this PR, you'll have to rebase with `main`.

**Comment by TiberiuGC** [comment]
> There are some additional commits in this PR, you'll have to rebase with `main`.

I think it's showing the full commit history since I'm using the same branch I've used for the improvement PR itself. 
```
eksctl % git fetch upstream 
eksctl % git merge upstream/main
Already up to date.
```

**Comment by TiberiuGC** [comment]
Just an extra note that we'd probably want to remove the `--wait` flag from cluster deletion after confirming it's not failing anymore.  Otherwise it mitigates most of the runtime improvement.

**Pull request merged**